daffodil-cli/src/templates/bash-template

```diff
@@ -73,4 +74,4 @@ then
 	CLASSPATH=$(cygpath -apw "$CLASSPATH")
 fi
 
-exec java $JOPTS -cp "$CLASSPATH" "$MAINCLASS" "$@"
+exec java -Dlog4j.configurationFile="$CONFDIR/log4j2.xml" $JOPTS -cp "$CLASSPATH" "$MAINCLASS" "$@"
```

Comment:
My thinking of using CONFDIR was that we might want to add additional configuration files in the future. This is only the configuration file for log4j, there might be others.

But I agree maybe we could do this better. Maybe we prepend additional java defines to JOPTS? Prepending allows user to be override the values. So for example, say we had multiple config files for different APIs, it might look something like this:

```bash
CONFDIR="$BINDIR/../conf/"
...
JOPTS="-Dlog4j.configurationFile=\"$CONFDIR/log4j2.xml\" $JOPTS"
JOPTS="-Dfoo.configurationFile=\"$CONFDIR/foo.xml\" $JOPTS"
JOPTS="-Dbar.configurationFile=\"$CONFDIR/bar.xml\" $JOPTS"

exec java $JOPTS -cp "$CLASSPATH" "$MAINCLASS" "$@"
```

That makes the exec line more clear and makes it obvious how to add new default java options if needed--just prepend them to JOPTS before we exec. Thoughts?

Reply:
Yes, I'm fine with defining the configuration file in JOPTS.  Let's do that.

Reply:
I've made the update to this file, but couldn't figure out a way to get it to work without requireing bash and arrays. I couldn't seem to get things to work correctly when things include spaces due to the quoting/expansion rules. If anyone knows any tricks to get it working with sh, that'd probably be preferred.